src/test/java/com/example/bookstore/repository/BookRepositoryTest.java

```diff
@@ -0,0 +1,40 @@
+package com.example.bookstore.repository;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import com.example.bookstore.model.Book;
+import com.example.bookstore.model.Category;
+import com.example.bookstore.repository.book.BookRepository;
+import java.math.BigDecimal;
+import java.util.Optional;
+import java.util.Set;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.jdbc.AutoConfigureTestDatabase;
+import org.springframework.boot.test.autoconfigure.orm.jpa.DataJpaTest;
+
+@DataJpaTest
+@AutoConfigureTestDatabase(replace = AutoConfigureTestDatabase.Replace.NONE)
+public class BookRepositoryTest {
+    @Autowired
+    private BookRepository bookRepository;
+
+    @Test
+    void findById_GivenExistingBook_ShouldReturnBook() {
```

Comment:
you should test your custom methods. Don’t test methods crud repository. Also better to use SQL for data preparing